common/src/jni/main/cpp/conscrypt/native_crypto.cpp

```diff
@@ -8044,7 +8062,7 @@ static jlong NativeCrypto_SSL_set_timeout(JNIEnv* env, jclass, jlong ssl_address
     }
 
     // Convert to seconds
-    long timeout = millis / 1000;
+    int64_t timeout = millis / 1000;
```

Comment:
Both are wrong. The API is a `uint32_t` and you should probably check the size or make it saturating. Though it used to be `long`, which is why it looks like this.